extension/chrome/settings/inbox/inbox.ts

```diff
@@ -72,14 +87,21 @@ export class InboxView extends View {
     ({ email_provider: emailProvider, picture: this.picture } = await AcctStore.get(this.acctEmail, ['email_provider', 'picture']));
     this.messageRenderer = await MessageRenderer.newInstance(this.acctEmail, this.gmail, this.relayManager, this.factory, this.debug);
     this.inboxNotificationModule.render();
+    this.preRenderSecureComposeInFullScreen();
+    if (Catch.isThunderbirdMail()) {
+      $('#container-gmail-banner').hide();
+    }
     try {
       await Settings.populateAccountsMenu('inbox.htm');
       if (emailProvider && emailProvider !== 'gmail') {
         $('body').text('Not supported for ' + emailProvider);
       } else {
         await this.inboxMenuModule.render();
+        const parsedThreadId = await this.parseThreadIdFromHeaderMessageId();
         if (this.threadId) {
           await this.inboxActiveThreadModule.render(this.threadId);
+        } else if (parsedThreadId) {
+          await this.inboxActiveThreadModule.render(parsedThreadId);
         } else {
           await this.inboxListThreadsModule.render(this.labelId);
```

Comment:
no need to parse thread id when we already have existing `threadId`, so `const parsedThreadId = await this.parseThreadIdFromHeaderMessageId();` should be moved to `else` case:
```ts
if (this.threadId) {
  await this.inboxActiveThreadModule.render(this.threadId);
} else {
  const parsedThreadId = await this.parseThreadIdFromHeaderMessageId();
  if (parsedThreadId) {
    await this.inboxActiveThreadModule.render(parsedThreadId);
  } else {
    await this.inboxListThreadsModule.render(this.labelId);
  }
}
```

Reply:
Ah yes, I see. It's illogical to always run the threadId parsing process when its known. I'll update it accordingly.